### PR TITLE
Do not retry if X-Airplane-Retryable=false

### PR DESCRIPTION
**Pull request by colinking**
This allows our API to indicate that a client should not retry, e.g. if the API knows an error is permanent.

### CODE REVIEW

**Comment by colinking** [comment]
Current dependencies on/for this PR:
* main
  * **PR #279** <a href="https://app.graphite.dev/github/pr/airplanedev/node-sdk/279" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/airplanedev/node-sdk/279?utm_source=stack-comment).

File: packages/lib/internal/fetcher.test.ts [hunk 1]
```diff
@@ -341,6 +341,33 @@ describe("Fetcher", () => {
...
+        )
+        .persist(true);
+      await expect(async () => {
+        await fetcher.post("/v0/throw/error");
+      }).rejects.toThrow(
+        new HTTPError("An unexpected error occurred", {
+          method: "POST",
+          url: "/v0/throw/error",
+          statusCode: 500,
+          // This should be retried b/c of the response header, even though it is a 500.
```
[the remark this answers is not in the document]
Comment (calufornia) [reply packages/lib/internal/fetcher.test.ts:365]: Side note: are these tests working? I checked this out locally, changed attempt to `10`, and the test still passed

**Review by calufornia — APPROVED** [review]

File: packages/lib/internal/fetcher.test.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (colinking) [reply packages/lib/internal/fetcher.test.ts:365]: Eek, good catch. Looks like this method only compares the error message, not the entire object. Will fix!

[the remark this answers is not in the document]
Comment (colinking) [reply packages/lib/internal/fetcher.test.ts:365]: Turns out this is a [shortcoming of jest](https://github.com/facebook/jest/issues/11693). I found a third-party package that does what we want: https://github.com/DanielHreben/jest-matcher-specific-error

**Pull request merged**